src/SIL.XForge.Scripture/ClientApp/src/xforge-common/jsonapi.service.ts

```diff
@@ -5,19 +5,20 @@ import Coordinator, {
 } from '@orbit/coordinator';
 import { Exception } from '@orbit/core';
 import {
-  buildQuery, ClientError, FindRecord, NetworkError, Operation, Query, QueryOrExpression, Record, RecordIdentity,
-  ReplaceRecordOperation, Schema, SchemaSettings, Transform, TransformOrOperations
+  buildQuery, ClientError, FindRecord, FindRecordsTerm, NetworkError, Operation, Query, QueryOrExpression, Record,
+  RecordIdentity, RecordRelationship, ReplaceRecordOperation, Schema, SchemaSettings, Transform, TransformOrOperations
 } from '@orbit/data';
 import IndexedDBSource from '@orbit/indexeddb';
 import IndexedDBBucket from '@orbit/indexeddb-bucket';
 import Store from '@orbit/store';
-import { clone, Dict } from '@orbit/utils';
+import { clone, Dict, eq, extend } from '@orbit/utils';
 import { ObjectId } from 'bson';
 import { from, fromEventPattern, merge, Observable } from 'rxjs';
 import { map, startWith } from 'rxjs/operators';
 
 import { XForgeJSONAPISource } from './jsonapi/xforge-jsonapi-source';
 import { LiveQueryObservable } from './live-query-observable';
+import { getResourceRefType, getResourceType, Resource, ResourceRef } from './models/resource';
 
 @Injectable({
   providedIn: 'root'
```

Comment:
JSONAPIService needs a good class comment to help developers know the value it provides and how it should be used.

Reply:
In fact this file is so critical to the success of our Offline capability I think we should consider carefully documenting all the public methods with some guidance on where and how they are expected to be used.